workitem/workitem_repository.go

```diff
@@ -690,48 +693,175 @@ func (r *GormWorkItemRepository) Fetch(ctx context.Context, spaceID uuid.UUID, c
+		r.Children = strings.TrimRight(r.Children, "}")
+		children := strings.Split(r.Children, ",")
+		childMap[r.IterationID] = children
+	}
 	countsMap := map[string]WICountsPerIteration{}
-	for _, iterationWithCount := range res {
-		countsMap[iterationWithCount.IterationID] = iterationWithCount
+	for _, i := range wiMap {
+		t := i.Total
+		c := i.Closed
+		if children, exists := childMap[i.IterationID]; exists {
+			for _, child := range children {
+				if _, exists := wiMap[child]; exists {
+					t += wiMap[child].Total
+					c += wiMap[child].Closed
+				}
+			}
+		}
+		countsMap[i.IterationID] = WICountsPerIteration{
+			IterationID: i.IterationID,
+			Total:       t,
+			Closed:      c,
+		}
 	}
 	return countsMap, nil
 }
 
 // GetCountsForIteration returns Closed and Total counts of WI for given iteration
 // It executes
 // SELECT count(*) as Total, count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed FROM "work_items" where fields@> concat('{"system.iteration": "%s"}')::jsonb and work_items.deleted_at is null
-func (r *GormWorkItemRepository) GetCountsForIteration(ctx context.Context, iterationID uuid.UUID) (map[string]WICountsPerIteration, error) {
+func (r *GormWorkItemRepository) GetCountsForIteration(ctx context.Context, itr *iteration.Iteration) (map[string]WICountsPerIteration, error) {
 	var res WICountsPerIteration
-	query := fmt.Sprintf(`SELECT count(*) as Total,
-						count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed
-						FROM "work_items"
-						where fields@> concat('{"system.iteration": "%s"}')::jsonb
-						and work_items.deleted_at is null`, iterationID)
-	db := r.db.Raw(query)
+	pathOfIteration := append(itr.Path, itr.ID)
+	// get child IDs of the iteration
+	var childIDs []uuid.UUID
+	iterationTable := iteration.Iteration{}
+	iterationTableName := iterationTable.TableName()
+	getIterationsOfSpace := fmt.Sprintf(`SELECT id FROM %s WHERE path <@ '%s' and space_id = '%s'`, iterationTableName, pathOfIteration.Convert(), itr.SpaceID.String())
+	db := r.db.Raw(getIterationsOfSpace)
+	db.Pluck("id", &childIDs)
+	if db.Error != nil {
+		log.Error(ctx, map[string]interface{}{
+			"path": pathOfIteration.Convert(),
+			"err":  db.Error,
+		}, "unable to fetch children for path")
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	childIDs = append(childIDs, itr.ID)
+
+	// build where clause usig above ID list
+	idsToLookFor := []string{}
+	for _, x := range childIDs {
+		idsToLookFor = append(idsToLookFor, fmt.Sprintf("'%s'", x.String()))
+	}
+	whereClause := strings.Join(idsToLookFor, ",")
+	query := fmt.Sprintf(`SELECT count(*) AS Total,
+						count(CASE fields->>'system.state'
+									WHEN 'closed' THEN '1'
+									ELSE NULL
+								END) AS Closed
+					FROM %s wi
+					WHERE fields->>'system.iteration' IN (%s)
+					AND wi.deleted_at IS NULL`,
+		workitemTableName, whereClause)
+	db = r.db.Raw(query)
```

Comment:
Same here, please use `?` instead of `%s` and don't use `fmt.Sprintf` for formatting SQL strings.